### PR TITLE
Fix Estimize processing issues that prevented two downloaders from completing

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

Prevents the `EstimizeEstimatesDataDownloader`, `EstimizeReleaseDataDownloader` types from crashing when encountering a ticker with a "defunct" indicator, but we are unable to determine the delimiter.

~Note: I think we should move this shared behavior into the abstract class both of the Downloaders inherit from to avoid code duplication. Willing to do it as part of this PR.~

Done 😃 
#### Related Issue

N/A
#### Motivation and Context

Ensures Estimize processes successfully
#### Requires Documentation Change

No
#### How Has This Been Tested?

Processed data for both `EstimizeEstimatesDataDownloader` and `EstimizeReleaseDataDownloader` successfully
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [ ] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by Martin-Molinero** [comment]
> Note: I think we should move this shared behavior into the abstract class both of the Downloaders inherit from to avoid code duplication. Willing to do it as part of this PR.

Sounds great 😄 and some unit tests for the parser method would be great too. Could be a public static as `NormalizeTicker()`

File: ToolBox/EstimizeDataDownloader/EstimizeReleaseDataDownloader.cs [hunk 1]
```diff
@@ -78,13 +78,15 @@ public override bool Run()
...
-                    var ticker = company.Ticker;
-                    if (ticker.IndexOf("defunct", StringComparison.OrdinalIgnoreCase) > 0)
+                    var estimizeTicker = company.Ticker;
+                    string ticker;
+
+                    if (!TryNormalizeDefunctTicker(estimizeTicker, out ticker))
                     {
-                        var length = ticker.IndexOf('-');
-                        ticker = ticker.Substring(0, length).Trim();
+                        Log.Error($"EstimizeReleaseDataDownloader(): Defunct ticker {estimizeTicker} is unable to be parsed. Continuing...");
```
Comment (Martin-Molinero) [inline ToolBox/EstimizeDataDownloader/EstimizeReleaseDataDownloader.cs:86]: missing `continue` statement

File: ToolBox/EstimizeDataDownloader/EstimizeDataDownloader.cs [hunk 2]
```diff
@@ -159,6 +164,39 @@ protected void SaveContentToFile(string destinationFolder, string ticker, IEnume
...
+                {
+                    var length = ticker.IndexOf(delimChar);
+
+                    // Continue until we exhaust all delimiters
+                    if (length == -1)
+                    {
+                        continue;
+                    }
+
+                    nonDefunctTicker = ticker.Substring(0, length).Trim();
```
Comment (Martin-Molinero) [inline ToolBox/EstimizeDataDownloader/EstimizeDataDownloader.cs:188]: Based on the tests, `Trim` isn't really required

Comment (gsalaz98) [reply ToolBox/EstimizeDataDownloader/EstimizeDataDownloader.cs:188]: I'll update the tests to include a `TICKER -defunct` case. I've seen tickers with a space between the defunct indicator before, hence the `.Trim()`

**Pull request merged**